src/metadata.rs

```diff
@@ -1,81 +1,431 @@
-        ) 
+    /// Generate a polars Expr that will do
+    /// an exact match on the MetricId
+    pub fn to_polars_expr(&self) -> Expr {
+        col(&self.to_col_name()).eq(self.to_query_string())
+    }
+
+    /// Generate a polars Expr that will generate
+    /// a fuzzy search for the content of the Id
+    pub fn to_fuzzy_polars_expr(&self) -> Expr {
+        col(&self.to_col_name())
+            .str()
+            .contains(lit(self.to_query_string()), false)
     }
 }
 
-// TODO we might want to just pass the MetricMetaData rather than
-// having to 
-impl From<MetricMetadata> for MetricRequest{
-    fn from(value: MetricMetadata) -> Self {
-        MetricRequest { 
-            column: value.parquet_column_name.clone(),
-            file: value.metric_parquet_file_url.clone().unwrap()
-         }
+impl Into<Expr> for MetricId {
+    fn into(self) -> Expr {
+        self.to_polars_expr()
     }
 }
 
-impl From<&MetricMetadata> for MetricRequest{
-    fn from(value: &MetricMetadata) -> Self {
-        MetricRequest { 
-            column: value.parquet_column_name.clone(), 
-            file: value.metric_parquet_file_url.clone().unwrap()  
+impl Default for CountryMetadataPaths {
+    fn default() -> Self {
+        Self {
+            // TODO move this to enviroment variable or config or something
+            base_url: "https://popgetter.blob.core.windows.net/popgetter-dagster-test/test_2"
+                .into(),
+            geometry: "geometry_metadata.parquet".into(),
+            metrics: "metric_metadata.parquet".into(),
+            country: "country_metadata.parquet".into(),
+            source_data: "source_data_releases.parquet".into(),
+            data_publishers: "data_publishers.parquet".into(),
         }
     }
 }
 
-pub fn load_metadata(path: &str) -> Result<SourceDataRelease> {
-    let file = File::open(path)?;
-    let reader = BufReader::new(file);
-    let release: SourceDataRelease = serde_json::from_reader(reader)?;
-    Ok(release)
+/// `CountryMetadataLoader` takes a country iso string
+/// along with a CountryMetadataPaths and provides methods
+/// for fetching and construting a `Metadata` catalouge.
+pub struct CountryMetadataLoader {
+    country: String,
+    paths: CountryMetadataPaths,
+}
+
+/// The metadata struct contains the polars `DataFrames` for
+/// the various different metadata tables. Can be constructed
+/// from a single `CountryMetadataLoader` or for all countries.
+/// It also provides the various functions for searching and
+/// getting `MetricRequests` from the catalouge.
+#[derive(Debug)]
+pub struct Metadata {
+    pub metrics: DataFrame,
+    pub geometries: DataFrame,
+    pub source_data_releases: DataFrame,
+    pub data_publishers: DataFrame,
+    pub countries: DataFrame,
+}
+
+impl Metadata {
+    /// If our metric_id is a regex, expand it in to a list of explicit `MetricIds`
+    pub fn expand_wildcard_metric(&self, metric_id: &MetricId) -> Result<Vec<MetricId>> {
```

Comment:
Is it worth renaming with `regex` instead of `wildcard`:
```suggestion
    pub fn expand_regex_metric(&self, metric_id: &MetricId) -> Result<Vec<MetricId>> {
```